### PR TITLE
Use DistrictSearchMultiSelectInput in Dref forms

### PR DESCRIPTION
**Pull request by shreeyash07**
## Changes
Replace MultiSelectInput with DistrictSeachMultiSelectInput in 
  - DrefApplicationForm
  - DrefFinalReportForm
  - DrefOperationalUpdateForm

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/DrefApplicationForm/Overview/index.tsx [hunk 1]
```diff
@@ -19,22 +21,23 @@ import Link from '#components/Link';
...
 } from '#utils/selectors';
 import useGlobalEnums from '#hooks/domain/useGlobalEnums';
 import useCountry from '#hooks/domain/useCountry';
 import NationalSocietySelectInput from '#components/domain/NationalSocietySelectInput';
 import CountrySelectInput from '#components/domain/CountrySelectInput';
 import DisasterTypeSelectInput from '#components/domain/DisasterTypeSelectInput';
+import DistrictSearchMultiSelectInput,
+{
+    type DistrictItem,
+} from '#components/domain/DistrictSearchMultiSelectInput';
```
Comment (frozenhelium) [inline src/views/DrefApplicationForm/Overview/index.tsx]: Let's avoid using prettier on file level
```suggestion
import DistrictSearchMultiSelectInput, {
    type DistrictItem,
} from '#components/domain/DistrictSearchMultiSelectInput';
```

File: src/views/DrefFinalReportForm/Overview/index.tsx [hunk 2]
```diff
@@ -20,21 +22,22 @@ import Button from '#components/Button';
...
 } from '#utils/selectors';
 import useGlobalEnums from '#hooks/domain/useGlobalEnums';
 import useCountry from '#hooks/domain/useCountry';
 import NationalSocietySelectInput from '#components/domain/NationalSocietySelectInput';
 import CountrySelectInput from '#components/domain/CountrySelectInput';
 import DisasterTypeSelectInput from '#components/domain/DisasterTypeSelectInput';
+import DistrictSearchMultiSelectInput,
+{
+    type DistrictItem,
+} from '#components/domain/DistrictSearchMultiSelectInput';
```
Comment (frozenhelium) [inline src/views/DrefFinalReportForm/Overview/index.tsx]:
```suggestion
import DistrictSearchMultiSelectInput, {
    type DistrictItem,
} from '#components/domain/DistrictSearchMultiSelectInput';
```

File: src/views/DrefOperationalUpdateForm/Overview/index.tsx [hunk 3]
```diff
@@ -36,6 +38,10 @@ import useCountry from '#hooks/domain/useCountry';
 import NationalSocietySelectInput from '#components/domain/NationalSocietySelectInput';
 import CountrySelectInput from '#components/domain/CountrySelectInput';
 import DisasterTypeSelectInput from '#components/domain/DisasterTypeSelectInput';
+import DistrictSearchMultiSelectInput,
+{
+    type DistrictItem,
+} from '#components/domain/DistrictSearchMultiSelectInput';
```
Comment (frozenhelium) [inline src/views/DrefOperationalUpdateForm/Overview/index.tsx]:
```suggestion
import DistrictSearchMultiSelectInput, {
    type DistrictItem,
} from '#components/domain/DistrictSearchMultiSelectInput';
```

**Review by frozenhelium — CHANGES_REQUESTED** [review]
Lets remove all unused variables
Please also do a `yarn lint` and `yarn typecheck` and fix all the issues before the PR.

**Review by tnagorra — APPROVED** [review]

**Pull request merged**